app/models/user.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  has_many :posts
+  attr_accessor :remember_token
+  after_create :remember
+  validates :name, presence: true
+  validates :email, presence: true, uniqueness: { case_sensitive: false }
+  has_secure_password
+
+  # Returns the hash digest of the given string.
+  def self.digest(string)
+    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
+    BCrypt::Password.create(string, cost: cost)
+  end
+
+  # Returns a random token.
+  def self.new_token
+    Digest::SHA1.hexdigest SecureRandom.urlsafe_base64.to_s
+  end
+
+  # Remembers a user in the database for use in persistent sessions.
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
```

Comment:
No need to call `User.digest` here. Directly use `remember_token`

Reply:
modified